Frontend/ui-driver/src/Screens/ReferralScreen/Handler.purs

```diff
@@ -26,14 +26,19 @@ import Screens.ReferralScreen.View as ReferralScreen
 import Types.App (FlowBT, GlobalState(..), REFERRAL_SCREEN_OUTPUT(..), ScreenType(..))
 import Screens.ReferralScreen.ScreenData as ReferralScreenData
 import Types.ModifyScreenState (modifyScreenState)
-import Data.Maybe(isJust)
+import Data.Maybe
 import Screens.Types (ReferralType(..))
 import Screens.ReferralScreen.ScreenData as ReferralScreenData
+import Merchant.Utils(getMerchant, Merchant(..))
 
 referralScreen:: FlowBT String REFERRAL_SCREEN_OUTPUT
 referralScreen = do
   (GlobalState state) <- getState
-  action <- lift $ lift $ runScreen $ ReferralScreen.screen state.referralScreen{ props{ stage = LeaderBoard } }
+  let currStage = if getMerchant unit == YATRIPARTNER then do
+                    if state.referralScreen.data.driverInfo.referralCode == Nothing then ComingSoonScreen
```

Comment:
@prakharritik instead of comparing it with nothing we can use the function isJust.

Reply:
Fixed.